bertopic/_bertopic.py

```diff
@@ -1058,15 +1087,120 @@ def _reduce_topics(self, documents: pd.DataFrame) -> pd.DataFrame:
+        #     documents = self._reduce_to_n_topics(documents)
+        # elif isinstance(self.nr_topics, str):
+        #     documents = self._auto_reduce_topics(documents)
+        # else:
+        #     raise ValueError("nr_topics needs to be an int or 'auto'! ")
 
         return documents
 
+    def _reduce_with_gmm(self, documents: pd.DataFrame) -> pd.DataFrame:
+        if not self.mapped_topics:
+            self.mapped_topics = dict()
+
+        initial_nr_topics = len(self.get_topics())
+
+        topic_vectors = self.topic_embeddings_doc
+        predictions = GaussianMixture(n_components=self.nr_topics,
+                                      random_state=0).fit_predict(topic_vectors)
+
+        merging_mapping = self._get_merging_mapping(predictions)  # mapping old and new topics
+        cluster_mapping = dict()
+
+        if self.mapped_topics is not None:
+            for prev_to_merge, prev_to_merge_into in self.mapped_topics.items():
+                # since we cannot access old topic embeddings, we need to map documents w.r.t. prev. mapping  # noqa
+                documents.loc[documents.Topic == prev_to_merge, "Topic"] = prev_to_merge_into
+
+        for cluster, topics_in_cluster in merging_mapping.items():
+            candidates_for_into = deepcopy(topics_in_cluster)
+
+            topic_to_merge_into = max(candidates_for_into, key=lambda x: self.get_topic_freq(x))
+            cluster_mapping[cluster] = topic_to_merge_into
+            topics_in_cluster.remove(topic_to_merge_into)  # topics to merge
+
+            for topic_to_merge in topics_in_cluster:
+                if topic_to_merge not in self.mapped_topics.keys():
+                    self.mapped_topics[topic_to_merge] = topic_to_merge_into
+                documents.loc[documents.Topic == topic_to_merge, "Topic"] = topic_to_merge_into
+
+        self._update_topic_size(documents)
+        self._extract_topics(documents)
+
+        cluster_predictions = [cluster_mapping[pred] for pred in predictions]
+        # update topic embeddings after reducing with clustering
+        # todo: we may use 'predictions' with iterating it with set, in the _create_topic_vectors()
+        self.topic_embeddings_doc = \
+            self._create_topic_vectors(method='doc',
+                                       embeddings=np.array(self.topic_embeddings_doc),
+                                       predictions=cluster_predictions)
+
+        nr_topics = len(self.get_topic_freq())
+        logger.info(f"Reduced number of topics from {initial_nr_topics} to {nr_topics}")
+
+        return documents
+
+    def _reduce_with_hdbscan(self, documents: pd.DataFrame, embeddings: np.ndarray) -> pd.DataFrame:
+        if not self.mapped_topics:
+            self.mapped_topics = dict()
+
+        initial_nr_topics = len(self.get_topics())
+
+        old_predictions = deepcopy(documents.Topic.values)
+        flat_clusterer = HDBSCAN_flat(embeddings, n_clusters=self.nr_topics,
```

Comment:
Update hdbscan version in Dockerfile. Check if the old models can be loaded in the new version.